typescript/lib/parsers/palm.ts

```diff
@@ -0,0 +1,50 @@
+import { DiscussServiceClient } from "@google-ai/generativelanguage";
+import { JSONObject } from "../../common";
+import { Prompt, Output } from "../../types";
+import { AIConfigRuntime } from "../config";
+import { InferenceOptions } from "../modelParser";
+import { ParameterizedModelParser } from "../parameterizedModelParser";
+import { GoogleAuth } from "google-auth-library";
+import { getAPIKeyFromEnv } from "../utils";
+
+/**
+ * Model Parser for PaLM Text Generation. PaLM API Currently does not support streaming
+ * models/chat-bison-001
+ */
+
+export class PaLMTextParser extends ParameterizedModelParser {
```

Comment:
nit: `PaLMTextGenerationParser` is more descriptive, but not a big deal

Reply:
naming convention comes from the models/api being referred to as PaLM Text and PaLM Chat